### PR TITLE
Feat: Added ConnectionCard component

### PR DESCRIPTION
**Pull request by RobinCsl**
Summary:
TODO Inquire why one of the tests for ConnectionCard fails in this environment and worked fine in @kiwicom/universal-components;
LocalizedPrice is not available in @kiwicom/universal-components@0.0.6

=> This must be revisited once we publish v0.0.8 of @kiwicom/universal-components

This is how it can be used:
```
const hasReturnConnection = true;
    const hasBadges = true;
    // const localizedPrice = '$455.3455';
    return (
      <ConnectionCard
        wayForth={[
          {
            id: 1,
            arrival: 'Berlin TXL',
            arrivalTime: '14:20',
            carrier: { code: 'OK', name: 'Czech Airlines' },
            tripDate: 'Mon 22 Oct',
            departure: 'Prague PRG',
            departureTime: '13:20',
            duration: '1h',
          },
          {
            id: 2,
            arrival: 'Moscow VKO',
            arrivalTime: '20:20',
            carrier: { code: 'FR', type: 'airline', name: 'Ryanair' },
            tripDate: 'Mon 22 Oct',
            departure: 'Berlin TXL',
            departureTime: '15:20',
            duration: '3h',
          },
        ]}
        wayBack={
          hasReturnConnection && [
            {
              id: 1,
              arrival: 'Prague PRG',
              arrivalTime: '14:00',
              carrier: { code: 'TO', name: 'Transavia France' },
              tripDate: 'Mon 25 Oct',
              departure: 'Moscow VKO',
              departureTime: '10:00',
              duration: '4h',
            },
          ]
        }
        duration={hasReturnConnection && '3 days'}
        {/* localizedPrice={localizedPrice} */}
        badges={
          hasBadges && [
            {
              id: 1,
              type: 'warning',
              children: 'Cheapest',
            },
            {
              id: 2,
              type: 'neutral',
              children: 'Wi-fi',
            },
          ]
        }
      />
    );
```

### CODE REVIEW

**Comment by RobinCsl** [comment]
Also, note that I replaced the original image consisting of two dots linked by a line with an arrow in the middle by RN implementation.

File: packages/components/src/ConnectionCard/ConnectionCard.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      arrivalTime={connection.arrivalTime}
+      carrier={connection.carrier}
+      tripDate={connection.tripDate}
+      departure={connection.departure}
+      departureTime={connection.departureTime}
+      duration={connection.duration}
+    />
+  ));
+
+const renderBadges = badges =>
```
Comment (tbergq) [inline packages/components/src/ConnectionCard/ConnectionCard.js]: `const RenderBadges = ({ badges }: Props) =>`

File: packages/components/src/ConnectionCard/ConnectionCard.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+              <View style={styles.leftShift}>
+                <Badge type="neutral">Returns in {duration}</Badge>
+              </View>
+            </ConnectionCardRow>
+          )}
+          <ConnectionCardRow>{renderTripSector(wayBack)}</ConnectionCardRow>
+        </React.Fragment>
+      )}
+      <ConnectionCardRow style={styles.lastRow}>
+        <BadgesContainer>{renderBadges(badges)}</BadgesContainer>
```
Comment (tbergq) [inline packages/components/src/ConnectionCard/ConnectionCard.js]: <BadgesContainer><RenderBadges badges={badges} /</BadgesContainer>

Comment (tbergq) [inline packages/components/src/ConnectionCard/ConnectionCard.js]:
```
<BadgesContainer>
<RenderBadges badges={badges} />
</BadgesContainer>
```

File: packages/components/src/ConnectionCard/TripSector.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          </View>
+          <View style={styles.tripItems}>
+            <View style={styles.time}>
+              <Text numberOfLines={1}>{departureTime}</Text>
+              <Text numberOfLines={1}>{arrivalTime}</Text>
+            </View>
+            <TimelineArrow />
+            <View style={styles.places}>
+              {this.renderText(departure)}
+              {this.renderText(arrival)}
```
Comment (tbergq) [inline packages/components/src/ConnectionCard/TripSector.js]: This does nothing a component does not

It is better to create a component. That is what react is for 🙂

**Comment by tbergq** [comment]
Also, I expect these components to change quite a bit when we connect them to relay

**Comment by RobinCsl** [comment]
@FilipMessa 🚀 ?

**Comment by RobinCsl** [comment]
Since you approved @FilipMessa, and I had to fix a conflict, I will merge it.

**Pull request merged**